### PR TITLE
Minor cleanups

### LINKED ISSUE #64: Disambiguate `num_orbitals`
### What is the expected enhancement?

@Cryoris brought up the very good suggestion, that we should make an effort to disambiguate the commonly used `num_orbitals` keyword argument. This is used throughout a large part of the stack with either of the two meanings:
1. number of *molecular* orbitals
2. number of *spin* orbitals

Before releasing Nature, we should attempt an effort to clearly disambiguate these cases by replacing the argument with (e.g.)
1. `num_molecular_orbitals`
2. `num_spin_orbitals`

This will clearly tell the user what is expected without having to look into the docstring of each class.

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

This PR is a collection of minor cleanups.
It also closes #64

### Details and comments

Grepping for `num_orbitals` still returns results in modules/classes which will be removed before the release. I did not bother changing things there, too. These include:
- `qiskit_nature.transformations`
- `qiskit_nature.components`

`qiskit_nature.algorithms` will be tackled during #83.

### CODE REVIEW

**Review by pbark — DISMISSED** [review]

**Comment by mrossinek** [comment]
Thanks for the approval. I think I will actually add some more things here, so I converted this to a draft for now.

File: qiskit_nature/mappers/second_quantization/__init__.py [hunk 1]
```diff
@@ -13,25 +13,9 @@
...
-from .direct_mapper import DirectMapper
 from .fermionic_mapper import FermionicMapper
 from .jordan_wigner_mapper import JordanWignerMapper
 from .linear_mapper import LinearMapper
-from .logarithmic_mapper import LogarithmicMapper
 from .parity_mapper import ParityMapper
 from .qubit_mapper import QubitMapper
 from .spin_mapper import SpinMapper
-
-__all__ = [
```
Comment (woodsp-ibm) [inline qiskit_nature/mappers/second_quantization/__init__.py:26]: Why was this completely removed? This is normal to have in the init https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces

Also I would expect this file will have to contain sphinx documentation that is similar to what you see in other init files. That still needs doing for this and others I imagine.

Comment (mrossinek) [reply qiskit_nature/mappers/second_quantization/__init__.py:26]: This is not handled consistently within Nature at the moment. I saw it was not there elsewhere and thus removed it.
I will make an effort to make this consistent instead.

I will also go through all `__init__` files and a `TODO(sphinx): documentation` comment which is easy to grep for when we get to the final preparation stages of this release. This is not something I am going to tackle in this PR.

**Comment by mrossinek** [comment]
I would also like to suggest replacing `indexes` with `indices` everywhere in the code base. What do you think?

File: qiskit_nature/mappers/second_quantization/__init__.py (hunk 1 above)
Comment (woodsp-ibm) [reply qiskit_nature/mappers/second_quantization/__init__.py:26]: Taking care of the code is one thing. Given we often allow the user to import modules from its parent then not all init files need docs. We will have to look over the docs anyway and sort things out which yes would be a separate PR later once things are a bit more ready. As a first pass it needs at least the navigation structure in place - then that can be filled out. Anyway its for later....

**Comment by mrossinek** [comment]
Besides my question above, I cannot think of any other pressing things which I would like to fix. Thus, this is now ready for review :+1:

**Comment by woodsp-ibm** [comment]
> I would also like to suggest replacing indexes with indices everywhere in the code base. What do you think?

Either are accepted as plural of index. If things are consistent everywhere I guess that would be good. If its in comments I would think its very low priority to change - for a param name etc., i.e. something that would end up in users code, then it would be good to have things consistent - do we have that where its a parameter name or something?

**Comment by mrossinek** [comment]
Both variants are being used in comments and as variable names (or parts thereof). `indices` appears to be the more widely used one.
[indexes.txt](https://github.com/Qiskit/qiskit-nature/files/6183818/indexes.txt)
[indices.txt](https://github.com/Qiskit/qiskit-nature/files/6183820/indices.txt)

**Comment by woodsp-ibm** [comment]
For internal variable names we can change it any time. Its more for param names on public API where we could expect the name to appear in user code. It would be nice to be consistent. I am fine with `indices` as that I think is more used in general

**Comment by mrossinek** [comment]
Although this is likely to not catch all cases a quick grep for `def.*indexes` and `def.*indices` returns only a single result for the former (a function of `UVCC`) and multiple for the latter as it is used in some args of private functions around the repo.

**Comment by mrossinek** [comment]
I tried a `s/indexes/indices/g` on Qiskit Nature... Let's see if the CI is still happy.
I also found some `DeprecationWarning`s being raised by the mappers because they used legacy methods from the Pauli operators. I fixed those, too.

**Review by pbark — APPROVED** [review]

**Comment by pbark** [comment]
Thanks @mrossinek for taking care of this!

**Comment by mrossinek** [comment]
I force pushed this branch because a mixture of online merges of the `master` branch and local ones confused this branch into thinking it included some others PRs. This branch is fixed now and contains the correct changes.
@pbark Please re-approve because this process dismissed your approval

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**